### PR TITLE
memory-usage-limit shouldn't disturb users

### PR DESCRIPTION
**Pull request by hicqu**
Signed-off-by: qupeng <qupeng@pingcap.com>

### What problem does this PR solve?

https://github.com/tikv/tikv/pull/10227 adds 2 new configurations into TiKV. It makes new learn cost for users.

### What is changed and how it works?

This PR does 2 things:
* make `memory-usage-high-water` be hidden for users, and default value changes to 0.9;
* calculate `memory-usage-limit` based on `block-cache.capacity`:
  so that users need only pay attation to the latter one.

With the change, memory quotas for different system memory capacity will be:
* system=8G, block-cache=3.6G, memory-usage-limit=6G, high-water=5.4G, page-cache=2G.
* system=16G, block-cache=7.2G, memory-usage-limit=12G, high-water=10.8G, page-cache=4G
* system=32G, block-cache=14.4G, memory-usage-limit=24G, high-water=21.6G, page-cache=8G

This PR also adds a new panel for raft entry evicting. Here is the capture for it:
图片

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- PR to update `pingcap/tidb-ansible`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test

### Release note 

```release-note
None
```

### CODE REVIEW

File: src/config.rs [hunk 1]
```diff
@@ -2564,26 +2559,43 @@ impl TiKvConfig {
...
+        }
+
+        let mut limit = self.memory_usage_limit.0.unwrap();
+        let total = ReadableSize(SysQuota::memory_limit_in_bytes());
+        if limit.0 > total.0 {
+            warn!(
+                "memory_usage_limit:{:?} > total:{:?}, fallback to total",
+                limit, total,
+            );
+            self.memory_usage_limit.0 = Some(total);
```
Comment (rleungx) [inline src/config.rs:2589]: Is it proper to fall back to total?

Comment (hicqu) [reply src/config.rs:2589]: Do you mean we should fallback to `default`? I thinks some users maybe want a value between `default` and `total`. So here if it exceeds `total`, fallback to `total` is better than `default`.

File: metrics/grafana/tikv_details.json [hunk 2]
```diff
@@ -13511,8 +13612,25 @@
...
           "id": 7236,
+          "legend": {
+            "show": false
+          },
           "links": [],
+          "repeat": null,
+          "repeatDirection": null,
+          "repeatedByRow": true,
+          "reverseYBuckets": false,
+          "scopedVars": {
```
Comment (rleungx) [inline metrics/grafana/tikv_details.json:13627]: Why change it?

Comment (hicqu) [reply metrics/grafana/tikv_details.json:13627]: I have no idea why it's changed. I will revert it.

File: src/config.rs [hunk 3]
```diff
@@ -2564,26 +2559,53 @@ impl TiKvConfig {
...
+            limit = total;
+        }
+
+        let default = Self::default_memory_usage_limit();
+        if limit.0 > default.0 {
+            warn!(
+                "memory_usage_limit:{:?} > default:{:?}, maybe page cache isn't enough",
```
Comment (zhangjinpeng1987) [inline src/config.rs:2606]: s/default/recommended size

Comment (hicqu) [reply src/config.rs]: Fixed.

File: components/raftstore/src/store/memory.rs [hunk 4]
```diff
@@ -51,9 +51,12 @@ pub fn needs_evict_entry_cache() -> bool {
...
         let ec_usage = MEMTRACE_ENTRY_CACHE.sum() as u64;
-        // Evict cache if entry cache memory usage reaches 1/3 of global,
-        // or 1GiB for small instances.
-        if ec_usage > GIB && (ec_usage > usage / 3 || usage <= 3 * GIB) {
+        // Evict if entry cache memory usage reaches 1/5 of global, or 1GiB for small instances.
+        // So for different system memory capacity, cache evict happens:
+        // * system=8G,  memory_usage_limit=6G,  evict=1.2G
+        // * system=16G, memory_usage_limit=12G, evict=2.4G
+        // * system=32G, memory_usage_limit=24G, evict=4.8G
+        if ec_usage > GIB && (ec_usage > usage / 5 || usage <= 5 * GIB) {
```
Comment (zhangjinpeng1987) [inline components/raftstore/src/store/memory.rs:59]: || usage >= 5GB?

File: components/raftstore/src/store/memory.rs [hunk 5]
```diff
@@ -51,9 +51,12 @@ pub fn needs_evict_entry_cache() -> bool {
...
         let ec_usage = MEMTRACE_ENTRY_CACHE.sum() as u64;
-        // Evict cache if entry cache memory usage reaches 1/3 of global,
-        // or 1GiB for small instances.
-        if ec_usage > GIB && (ec_usage > usage / 3 || usage <= 3 * GIB) {
+        // Evict if entry cache memory usage reaches 1/5 of global, or 1GiB for small instances.
+        // So for different system memory capacity, cache evict happens:
+        // * system=8G,  memory_usage_limit=6G,  evict=1.2G
+        // * system=16G, memory_usage_limit=12G, evict=2.4G
+        // * system=32G, memory_usage_limit=24G, evict=4.8G
+        if ec_usage > GIB && ec_usage > usage / 5 {
```
Comment (zhangjinpeng1987) [inline components/raftstore/src/store/memory.rs:59]: How about the total memory is 128GB? I prefer 'if ec_usage > GIB && (ec_usage > usage/5 || ec_usage >= 5GB)'

**Review by zhangjinpeng1987 — APPROVED** [review]
LGTM

**Review by rleungx — APPROVED** [review]

**Pull request merged**